python/packages/autogen-ext/src/autogen_ext/models/azure/_azure_ai_client.py

```diff
@@ -0,0 +1,406 @@
+    """
+    if not re.match(r"^[a-zA-Z0-9_-]+$", name):
+        raise ValueError(f"Invalid name: {name}. Only letters, numbers, '_' and '-' are allowed.")
+    if len(name) > 64:
+        raise ValueError(f"Invalid name: {name}. Name must be less than 64 characters.")
+    return name
+
+
+class AzureAIChatCompletionClient(ChatCompletionClient):
+    def __init__(self, **kwargs: Unpack[AzureAIChatCompletionClientConfig]):
+        if "endpoint" not in kwargs:
+            raise ValueError("endpoint is required for AzureAIChatCompletionClient")
+        if "credential" not in kwargs:
+            raise ValueError("credential is required for AzureAIChatCompletionClient")
+        if "model_capabilities" not in kwargs:
+            raise ValueError("model_capabilities is required for AzureAIChatCompletionClient")
+        if _is_github_model(kwargs['endpoint']) and "model" not in kwargs:
+            raise ValueError("model is required for when using a Github model with AzureAIChatCompletionClient")
+
+        # TODO: Change
+        _endpoint = kwargs.pop("endpoint")
+        _credential = kwargs.pop("credential")
+        self._model_capabilities = kwargs.pop("model_capabilities")
+        self._create_args = kwargs.copy()
+
+        self._client = ChatCompletionsClient(_endpoint, _credential, **self._create_args)
+        self._actual_usage = RequestUsage(prompt_tokens=0, completion_tokens=0)
+        self._total_usage = RequestUsage(prompt_tokens=0, completion_tokens=0)
+
+    async def create(
+        self,
+        messages: Sequence[LLMMessage],
+        tools: Sequence[Tool | ToolSchema] = [],
+        json_output: Optional[bool] = None,
+        extra_create_args: Mapping[str, Any] = {},
+        cancellation_token: Optional[CancellationToken] = None,
+    ) -> CreateResult:
+        extra_create_args_keys = set(extra_create_args.keys())
+        if not create_kwargs.issuperset(extra_create_args_keys):
+            raise ValueError(f"Extra create args are invalid: {extra_create_args_keys - create_kwargs}")
+
+        # Copy the create args and overwrite anything in extra_create_args
+        create_args = self._create_args.copy()
+        create_args.update(extra_create_args)
+
+        if self.capabilities["vision"] is False:
+            for message in messages:
+                if isinstance(message, UserMessage):
+                    if isinstance(message.content, list) and any(isinstance(x, Image) for x in message.content):
+                        raise ValueError("Model does not support vision and image was provided")
+
+        if json_output is not None:
+            if self.capabilities["json_output"] is False and json_output is True:
+                raise ValueError("Model does not support JSON output")
+
+            if json_output is True:
+                create_args["response_format"] = ChatCompletionsResponseFormatJSON()
```

Comment:
I think we need to check existing `response_format` value to make sure we aren't overwriting it.